packages/x-date-pickers/src/models/adapters.ts

```diff
@@ -0,0 +1,586 @@
+   * @param {TDate} comparing The date to compare with the reference date.
+   * @returns {boolean} `true` if the reference date is after the second date.
+   */
+  isAfter(value: TDate, comparing: TDate): boolean;
+  /**
+   * Returns `true` if the day of the reference date is after the day of the second date.
+   * TODO v7: Consider adding a `unit` param to `isAfter` and drop this method.
+   * @template TDate
+   * @param {TDate} value The reference date.
+   * @param {TDate} comparing The date to compare with the reference date.
+   * @returns {boolean} `true` if the day of the reference date is after the day of the second date.
+   */
+  isAfterDay(value: TDate, comparing: TDate): boolean;
+  /**
+   * Returns `true` if the year of the reference date is after the year of the second date.
+   * TODO v7: Consider adding a `unit` param to `isAfter` and drop this method.
+   * @template TDate
+   * @param {TDate} value The reference date.
+   * @param {TDate} comparing The date to compare with the reference date.
+   * @returns {boolean} `true` if the year of the reference date is after the year of the second date.
+   */
+  isAfterYear(value: TDate, comparing: TDate): boolean;
+  /**
+   * Returns `true` if the reference date is before the second date.
+   * @template TDate
+   * @param {TDate} value The reference date.
+   * @param {TDate} comparing The date to compare with the reference date.
+   * @returns {boolean} `true` if the reference date is before the second date.
+   */
+  isBefore(value: TDate, comparing: TDate): boolean;
+  /**
+   * Returns `true` if the day of the reference date is before the day of the second date.
+   * TODO v7: Consider adding a `unit` param to `isBefore` and drop this method.
+   * @template TDate
+   * @param {TDate} value The reference date.
+   * @param {TDate} comparing The date to compare with the reference date.
+   * @returns {boolean} `true` if the day of the reference date is before the day of the second date.
+   */
+  isBeforeDay(value: TDate, comparing: TDate): boolean;
+  /**
+   * Returns `true` if the year of the reference date is before the year of the second date.
+   * TODO v7: Consider adding a `unit` param to `isBefore` and drop this method.
+   * @template TDate
+   * @param {TDate} value The reference date.
+   * @param {TDate} comparing The date to compare with the reference date.
+   * @returns {boolean} `true` if the year of the reference date is before the year of the second date.
+   */
+  isBeforeYear(value: TDate, comparing: TDate): boolean;
+  /**
+   * Returns `true` if the value is withing the provided range.
+   * @template TDate
+   * @param {TDate} value The value to test.
+   * @param {[TDate, TDate]} range The range in which the value should be.
+   * @returns {boolean} `true` if the value is withing the provided range.
+   */
+  isWithinRange(value: TDate, range: [TDate, TDate]): boolean;
+  /**
+   * Return the start of the year for the given date.
```

Comment:
Isn't it technically a bit incorrect? 🤔 
This and the following methods would return the given date with a certain time unit set to `start` or `end` of it.

Reply:
I took date-fns description here https://date-fns.org/v2.29.3/docs/startOfYear

Why is it incorrect for you ?
If I run it with today's date, it will return `2023-01-01T00:00`, which for me matches the description.

Reply:
Maybe it's just me, but given this description, I wouldn't know what kind of response should I expect from this method. Is it a date? Or is it a year number?
Anyways, maybe I'm looking at it with a bit too critical eyes. If everyone's happy with this description on `date-fns`, it's gonna be fine for us as well. 👍 😆